### DISCUSSION TITLE
CSLA 8 - Setting `InputProperties` in the ctor of a BusinessRule is throwing an error

### DISCUSSION THREAD

**Discussion by swegele** [MarimerLLC/csla#3900] [Questions] [answered]
After upgrading to CSLA 8 - I'm getting a weird error on some broken rule classes in their CTOR.
Don't get distracted by the fact that I'm not doing anything with the list...I stripped it down to it's bare minimum in testing and it is still throwing the error.  So it seems that even an attempt to set InputProperties to a new list is throwing the error.
This code was fine in .net 6 and CSLA 6.

```
{"DataPortal.Fetch failed (Method not found: 'Void Csla.Rules.BusinessRuleBase.set_InputProperties(System.Collections.Generic.List`1<Csla.Core.IPropertyInfo>)'.)"}
```

Here is an example that throws the error:

```csharp
    public class NoLeadingSpacesRule : BusinessRule
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="NoLeadingSpacesRule"/> class.
        /// </summary>
        /// <param name="primaryProperty">
        /// The string property to test
        /// </param>
        public NoLeadingSpacesRule(PropertyInfo<string> primaryProperty)
            : base(primaryProperty)
        {
            InputProperties = new List<IPropertyInfo>();
        }

        protected override void Execute(IRuleContext context)
        {
            string message = string.Format("{0} must not have leading space(s)", PrimaryProperty.FriendlyName);
            string value = (string)ReadProperty(context.Target, PrimaryProperty);

            if (!string.IsNullOrWhiteSpace(value) && value.StartsWith(" "))
            {
                context.AddErrorResult(message);
            }
        }
    }

```

Two really weird things:
- If I remove the `InputProperties = new List<IPropertyInfo>();` it works fine.
- With the line there I cannot even hit a breakpoint in the constructor.  But removing the line then I can hit breakpoints in the constructor.

**Comment by ossendorf-at-hoelscher** [comment, marked as answer]
That sounds really weird. We are at 8.1.0 and are setting the `InputProperties`, too without any problems.
Are you sure you have the right assemblies are loaded? For v9 I removed the set on that list: #3880

**Reply by swegele** [reply]
Thanks for chiming in.  If it is working for you, then I will assume it is a side effect of the way I am running the project right now (copying locally built files over the nuget dlls) as we work on figuring out [that other bug](https://github.com/MarimerLLC/csla/discussions/3815).
